src/worker/main.ts

```diff
@@ -174,12 +175,21 @@ const sendQueueMetrics = async () => {
 
 const commonMiddleware = (jobHandler) => sentryMiddleware(setDelayOnRateLimiting(jobHandler));
 
+const randomInteger = (min, max) => {
+	return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
 export const start = (): void => {
 	initializeSentry();
 
 	// exposing queue metrics at a regular interval
 	setInterval(sendQueueMetrics, 60000);
 
+	// Run the deduplication job every 8 to 12 minutes (spreading it out a bit to make it less likely that two nodes are
+	// running it at exactly the same time). It should be less than 15 minutes, because if the installation queue is
+	// blocked for 15 minutes, sync jobs will begin to go into the "stalled" state.
+	setInterval(deduplicateInstallations, randomInteger(8 * 60 * 1000, 12 * 60 * 1000));
```

Comment:
Is this run per node instance? It that's one per core per ec2 instance, that can be quite a lot. Can we e.g. add an FF in the dedupe method, making it a no-op with X% probability, meaning we can scale the risk of concurrent dedupes